the-blue-alliance-ios/Core Data/Match.swift

```diff
@@ -79,91 +92,105 @@ extension Match: Managed {
         return dateFormatter.string(from: date)
     }
 
+    /**
+     Returns the alliance with an allianceKey of 'red'.
+    */
+    var redAlliance: MatchAlliance? {
+        return alliance(with: "red")
+    }
+
+    /**
+     Returns the trimmed team keys for the red alliance.
+     */
+    var redAllianceTeamNumbers: [String] {
+        return redAlliance?.teams.map({ Team.trimFRCPrefix($0) }).reversed() ?? []
+    }
+
+    /**
+     Returns the alliance with an allianceKey of 'blue'.
+     */
+    var blueAlliance: MatchAlliance? {
+        return alliance(with: "blue")
+    }
+
+    /**
+     Returns the trimmed team keys for the blue alliance.
+     */
+    var blueAllianceTeamNumbers: [String] {
+        return blueAlliance?.teams.map({ Team.trimFRCPrefix($0) }).reversed() ?? []
+    }
+
+    private func alliance(with allianceKey: String) -> MatchAlliance? {
+        return (alliances?.allObjects as? [MatchAlliance])?.first(where: { $0.allianceKey == allianceKey })
+    }
+
     @discardableResult
     static func insert(with model: TBAMatch, for event: Event, in context: NSManagedObjectContext) -> Match {
         let predicate = NSPredicate(format: "key == %@", model.key)
         return findOrCreate(in: context, matching: predicate) { (match) in
             // Required: compLevel, eventKey, key, matchNumber, setNumber
             match.key = model.key
-            match.compLevel = model.compLevel
-
-            let compLevelStruct = MatchCompLevel(rawValue: match.compLevel!)
-            match.compLevelInt = compLevelStruct!.intVal
+            match.compLevelString = model.compLevel
 
             match.setNumber = Int16(model.setNumber)
             match.matchNumber = Int16(model.matchNumber)
 
-            // TODO: Think about converting this Alliance stuff in to an Alliance object, like we do in the API
-            // It might actually make sense - we can store an `MatchAlliance` that can have a key associated with it too
-            // That way, when we pull `winningAlliance`, it can be more dynamic
-            if let redAlliance = model.redAlliance {
-                match.redAlliance = NSMutableOrderedSet(array: redAlliance.teams.map({ (teamKey) -> Team in
-                    return Team.insert(withKey: teamKey, in: context)
-                }))
-                if redAlliance.score > -1 {
```

Comment:
This needs to come back - matches that haven't been played yet have scores of -1